Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-87.ca-central-1.compute.internal
Sha: 06f04e324812d1970ba359078d2860ed8ee08128
Number of Error(s): 8
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-14491/HPCC-Platform/esp/smc/SMCLib/TpWrapper.cpp:2209:1: error: no return statement in function returning non-void [-Werror=return-type]
cc1plus: some warnings being treated as errors
make[2]: *** [esp/smc/SMCLib/CMakeFiles/SMCLib.dir/build.make:102: esp/smc/SMCLib/CMakeFiles/SMCLib.dir/TpWrapper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:12864: esp/smc/SMCLib/CMakeFiles/SMCLib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-01-27_14-59-40
time logs started
Prep time   :     7 sec (00:00:07)
Build time  :   479 sec (00:07:59)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   486 sec (00:08:06)
time logs end

[Build error:](https://gist.github.com/b096004cdfa415e8f5cebd4fa1bdce77)
Build log [Build log](https://gist.github.com/HPCCSmoketest/b096004cdfa415e8f5cebd4fa1bdce77#file-result-21-01-27-14-51-20-log)

--- a/esp/services/ws_topology/ws_topologyService.cpp
+++ b/esp/services/ws_topology/ws_topologyService.cpp
@@ -1397,9 +1397,9 @@ bool CWsTopologyEx::onTpClusterInfo(IEspContext &context, IEspTpClusterInfoReque
         Owned<IRemoteConnection> conn = querySDS().connect("/Status/Servers/", myProcessSession(),RTM_SUB,SDS_LOCK_TIMEOUT);
         if (conn)
         {
-            Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(req.getName());
+            Owned<IConstWUClusterInfo> clusterInfo = getWUClusterInfoByName(req.getName());
             if (clusterInfo == nullptr)
-                throw MakeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "Invalid Target Cluster name provided: '%s'", req.getName());
+                throw makeStringExceptionV(ECLWATCH_INVALID_CLUSTER_NAME, "Invalid Target Cluster name provided: '%s'", req.getName());
 
             SCMStringBuffer thorQueues;
             clusterInfo->getThorQueue(thorQueues);

--- a/esp/services/ws_workunits/ws_workunitsHelpers.cpp
+++ b/esp/services/ws_workunits/ws_workunitsHelpers.cpp
@@ -31,6 +31,7 @@
 #include "hqlexpr.hpp"
 #include "rmtsmtp.hpp"
 #include "LogicFileWrapper.hpp"
+#include "TpWrapper.hpp"
 
 #ifndef _NO_LDAP
 #include "ldapsecurity.ipp"
@@ -1290,11 +1291,7 @@ bool WsWuInfo::getClusterInfo(IEspECLWorkunit &info, unsigned long flags)
     {
         int clusterTypeFlag = 0;
 
-        Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
-        Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
-        Owned<IPropertyTree> root = &constEnv->getPTree();
-
-        Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName.str());
+        Owned<IConstWUClusterInfo> clusterInfo = getWUClusterInfoByName(clusterName.str());
         if (clusterInfo.get())
         {//Set thor flag or roxie flag in order to display some options for thor or roxie
             ClusterType platform = clusterInfo->getPlatform();

--- a/esp/smc/SMCLib/TpWrapper.cpp
+++ b/esp/smc/SMCLib/TpWrapper.cpp
@@ -2193,6 +2193,21 @@ extern TPWRAPPER_API unsigned getContainerWUClusterInfo(CConstWUClusterInfoArray
     return clusters.ordinality();
 }
 
+extern TPWRAPPER_API IConstWUClusterInfo* getWUClusterInfoByName(const char* clusterName)
+{
+#ifndef _CONTAINERIZED
+    return getTargetClusterInfo(clusterName);
+#else
+    VStringBuffer xpath("queues[@name='%s']", clusterName);
+    IPropertyTree* queue = queryComponentConfig().queryPropTree(xpath);
+    if (!queue)
+        return nullptr;
+
+    return new CContainerWUClusterInfo(queue->queryProp("@name"), queue->queryProp("@type"),
+        (unsigned) queue->getPropInt("@width", 1));
+#endif
+}
+
 extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSocketFactory>& connMap)
 {
     Owned<IPropertyTreeIterator> services = queryComponentConfig().getElements("services[@type='roxie']");

--- a/esp/smc/SMCLib/TpWrapper.hpp
+++ b/esp/smc/SMCLib/TpWrapper.hpp
@@ -213,6 +213,7 @@ private:
 extern TPWRAPPER_API ISashaCommand* archiveOrRestoreWorkunits(StringArray& wuids, IProperties* params, bool archive, bool dfu);
 
 extern TPWRAPPER_API unsigned getContainerWUClusterInfo(CConstWUClusterInfoArray& clusters);
+extern TPWRAPPER_API IConstWUClusterInfo* getWUClusterInfoByName(const char* clustName);
 
 extern TPWRAPPER_API IStringIterator *getContainerTargetClusters(const char* processType, const char* processName);
 extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSocketFactory>& connMap);

